interfaces/repo.go

```diff
@@ -267,10 +267,15 @@ func (r *Repository) RemoveSlot(snapName, slotName string) error {
 
 // Connect establishes a connection between a plug and a slot.
 // The plug and the slot must have the same interface.
-func (r *Repository) Connect(plugSnapName, plugName, slotSnapName, slotName string) error {
+func (r *Repository) Connect(ref *ConnRef) error {
```

Comment:
The original thinking behind PlugRef and SlotRef is that they'd be more like values than references of an object with an identity. I think we should use the same pattern here: ConnRef is a value without an underlying identity, dropping the *s and &s everywhere.